tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -1824,8 +1824,9 @@ void SwiftEditorDocument::parse(ImmutableTextSnapshotRef Snapshot,
   // all tokens are visited and thus token collection is invalid
   CompInv.getLangOptions().CollectParsedToken = (SyntaxCache == nullptr);
   // Access to Impl.SyntaxInfo is guarded by Impl.AccessMtx
+  std::string ResolvedPath = SwiftLangSupport::resolvePathSymlinks(Impl.FilePath);
```

Comment:
This is problematic because syntactic requests need to be as fast as possible (real-time, as-you-type performance) but `realpath` is notoriously slow.
I don't quite fully understand the issue so not sure what else to suggest, but is there something else that can be done?